src/uwtools/api/template.py

```diff
@@ -11,11 +11,12 @@
 
 
 def render(
-    values: Union[dict, Path],
+    values_src: Optional[Union[dict, Path]] = None,
```

Comment:
I couldn't see a reason why users shouldn't be able to provide any combination of `dict`- or file-based values and/or CLI overrides and/or environment-variable values. In the extreme case, they could provide none of these, in which case rendering a template with Jinja2 expressions would be an error *except* if `--partial` is supplied, in which case the template passes through unmodified -- maybe not very useful, but who knows?

I also renamed `values` to `values_src` because I wanted to use `values` to represent the final set of values from the `dict`/file source plus overrides plus environment variables.